test/integration/test_integration_basics.py

```diff
@@ -33,7 +31,7 @@ def test_http_h1(http_test_server_fixture):
   assertCounterEqual(counters, "upstream_rq_pending_total", 1)
   assertCounterEqual(counters, "upstream_rq_total", 25)
   assertCounterEqual(counters, "default.total_match_count", 1)
-  assertEqual(len(counters), 14)
+  assertEqual(len(counters), 12)
```

Comment:
Note: we expect less counters, because we grab the statistics before we shutdown the pools.
This implies that we won't observe some connections being closed, which I think makes sense.